src/storage/src/source/reclock.rs

```diff
@@ -185,29 +186,64 @@ where
         // We need to compute the answer by iterating over the consolidated remap trace which will
         // present to us one diff at a time. We know that by construction at any given IntoTime
         // time the remap collection accumulates to a well formed antichain. That is, it contains
-        // exactly one copy of mutually incomparable FromTime elements. Therefore, when we see a
-        // positive diff of some element `from_t` occuring at some timestamp `into_t` we also know
-        // that the FromTime frontier at `into_t` included that element.
+        // exactly one copy of mutually incomparable FromTime elements.
         //
-        // What we do then is that for every positive diff at IntoTime `ts` we check whether the
-        // FromTime element `frontier_ts` is less than or equal to the timestamp to be reclocked
-        // `src_ts`:
-        // * If it is, then the `FromTime` frontier at `ts` *must* be less than or equal to
-        //   `src_ts` and so we know that `src_ts` should *not* be reclocked at `ts` because it is
-        //   beyond the frontier. We codify that by adding `ts` into a mutable antichain with the
-        //   minimum frequency.
-        // * If it is not, then the `FromTime` frontier at `ts` *may* be less than or equal to
-        //   `src_ts` so `src_ts` *may* be reclocked to `ts`. We codify that by adding `ts` into a
-        //   mutable antichain with a frequency of 1.
+        // We also know that `src_ts` is beyond the since frontier, therefore there exist witness
+        // timestamps `from_ts` that are less than or equal to `src_ts` and occur at IntoTime times
+        // with a positive diff. We also know that `src_ts` is not beyond the upper frontier,
+        // therefore all the positive diffs of the witness times must be retracted at subsequent
+        // IntoTime times.
         //
-        // Once we have considered all the diffs we will end up with a mutable antichain that
-        // contains timestamps with frequency 1 for all the IntoTimes times that `src_ts` might be
-        // reclocked to and timestamps with frequency Diff::MIN for all the IntoTimes that `src_ts`
-        // definitely is not reclocked to.
+        // This cycle may be repeated an arbitrary amount of times until the final retraction. The
```

Comment:
is "cycle" the write word here? its most like "these facts may be repeated

Reply:
maybe "cycle" or "pattern. I'd also maybe change "final" -> "last" or "last such"